sql/core/src/test/scala/org/apache/spark/sql/CollationSuite.scala

```diff
@@ -32,6 +33,11 @@ import org.apache.spark.sql.types.StringType
 class CollationSuite extends DatasourceV2SQLBase {
   protected val v2Source = classOf[FakeV2ProviderWithCustomSchema].getName
 
+  protected override def beforeAll(): Unit = {
+    System.setProperty(IS_TESTING.key, "true")
```

Comment:
We have already set the SQL config to `Utils.isTesting`. This should not be necessary at all.

Reply:
Previously this Suite extended some other class which did not set this key to true, new merges created a conflict for me. Will remove it.

Reply:
Actually, I have just checked, `DatasourceV2SQLBase` extends `QueryTest` and is implementing `SharedSparkSession`. `QueryTest` does not implement `beforeAll` and `SharedSparkSession` does, but without setting IS_TESTING key.

Reply:
all spark test suites should be a subclass of `SparkFunSuite` which sets this flag.

Reply:
I am not sure what happens here, but I would presume that `SharedSparkSession` overrules the `SparkFunSuite` as the former is used as an interface after the latter. What I want to say is that without me explicitly saying this part of the code, IS_TESTING is not set to true and I suspect it is because of the `SharedSparkSession` which does not set that key.

Reply:
Do you maybe have some insight in ordering of class/trait extensions in Scala?

Reply:
I tried this and it prints true correctly
```
  test("aa") {
    println("##########    " + Utils.isTesting)
  }
```

can you double check?

Reply:
I have just debugged without `CollationSuite.beforeAll()` override. Order of calling was `SharedSparkSession.beforeAll()` -> `SQLConf.COLLATION_ENABLED` setting -> `SparkFunSuite.beforeAll()`. This is causing the proposed test to write `true` even though it was not set for the actual purpose and that is before SQLConf invocation, so that the desired config is set with the right value. Could you check that compilation does the same for you? I am not sure why the compilation proceeds in the following manner, but if needed I could investigate more?

Reply:
What's wrong with this order of calling? `Utils.isTesting` returns true properly and we are fine?